tests/ZeroC.Slice.Tests/CustomTypeTests.cs

```diff
@@ -61,4 +246,14 @@ public static MyCustomType DecodeCustomType(this ref SliceDecoder decoder)
         myCustom.Value = decoder.DecodeInt32();
         return myCustom;
     }
+
+    public static MyCustomType? DecodeNullableCustomType(this ref SliceDecoder decoder)
+    {
+        switch(decoder.DecodeString())
```

Comment:
You should use a `switch` expression instead

```
return decoder.DecodeString() switch
{
    "nope" => null,
    "yep!" => (MyCustomType?)decoder.DecodeCustomType(),
    _ => throw new InvalidDataException("decoded invalid custom type"),
};
```

Reply:
Updated to a switch expression.